priv/cabbage/apps/itest/test/itest/deposits_test.exs

```diff
@@ -19,11 +19,16 @@ defmodule DepositsTests do
   alias Itest.Account
   alias Itest.ApiModel.WatcherSecurityCriticalConfiguration
   alias Itest.Client
+  alias Itest.Reorg
   alias Itest.Transactions.Currency
 
   setup do
+    Reorg.finish_reorg()
+
     [{alice_account, alice_pkey}, {bob_account, _bob_pkey}] = Account.take_accounts(2)
 
+    Reorg.start_reorg()
```

Comment:
also, another idea. Perhaps theres a better interface and explanation to the developer what is going on.
What if we tie the whole reorg period into a anonymous fn?

this way, the developer knows which parts are going to be under stress of the chain reorg? wdyt?

this would cause more changes to the codebase, but maybe it would be more evident where a reorg is happening and when it's finished. so you could expose a particular block of code to reorg events...

Reply:
I tried to make it like
```elixir
Reorg.start_reorg(fn ->
...
end)
```
But it doesn't seem to work: reorgs just doesn't have enough time to be triggered. Because the anonymous function finishes its execution faster than pausing/unpausing operations are performed. 
I think it's logical to leave it in the setup callback.

Reply:
Perhaps the `start_reorg` function that executes the anonymous function can wait until reorgs start to appear, and then execute it's anonymous function that comes in as argument?

Reply:
@InoMurko It's not obvious to me what to wrap in this anonymous function - defgiven, defwhen, defthen?

Reply:
aha! lemme give you an example...

Reply:
so for example (looking at deposits_test.exs):
```
defwhen ~r/^Alice deposits "(?<amount>[^"]+)" ETH to the root chain$/,
          %{amount: amount},
          %{alice_account: alice_account} = state do
    initial_balance = Itest.Poller.root_chain_get_balance(alice_account)
Itest.Reorg.do_it(fn-> 
    {:ok, receipt_hash} =
      amount
      |> Currency.to_wei()
      |> Client.deposit(alice_account, Itest.PlasmaFramework.vault(Currency.ether()))
end)
    gas_used = Client.get_gas_used(receipt_hash)

    {_, new_state} =
      Map.get_and_update!(state, :gas, fn current_gas ->
        {current_gas, current_gas + gas_used}
      end)

    balance_after_deposit = Itest.Poller.root_chain_get_balance(alice_account)

    state = Map.put_new(new_state, :alice_ethereum_balance, balance_after_deposit)
    {:ok, Map.put_new(state, :alice_initial_balance, initial_balance)}
  end
```

Reply:
this way I would make sure a deposit happens on one of the blocks that gets reorged, and I prove that the childchain and watchers detect the deposit on the correct block thats also reorg protected (because the reorg is already over!)

Reply:
@InoMurko I've done exactly that for deposits tests. Please take a look.